include/kernel.h

```diff
@@ -371,12 +377,50 @@ typedef void (*k_thread_entry_t)(void *p1, void *p2, void *p3);
  *
  * @return ID of new thread.
  */
-extern k_tid_t k_thread_spawn(char *stack, size_t stack_size,
+extern __deprecated k_tid_t k_thread_spawn(char *stack, size_t stack_size,
 			      k_thread_entry_t entry,
 			      void *p1, void *p2, void *p3,
 			      int prio, u32_t options, s32_t delay);
 
 /**
+ * @brief Create a thread.
+ *
+ * This routine initializes a thread, then schedules it for execution.
+ *
+ * The new thread may be scheduled for immediate execution or a delayed start.
+ * If the newly spawned thread does not have a delayed start the kernel
+ * scheduler may preempt the current thread to allow the new thread to
+ * execute.
+ *
+ * Thread options are architecture-specific, and can include K_ESSENTIAL,
+ * K_FP_REGS, and K_SSE_REGS. Multiple options may be specified by separating
+ * them using "|" (the logical OR operator).
+ *
+ * Historically, users often would use the beginning of the stack memory region
+ * to store the struct k_thread data, although corruption will occur if the
+ * stack overflows this region and stack protection features may not detect this
+ * situation.
+ *
+ * @param new_thread Pointer to uninitialized struct k_thread
+ * @param stack Pointer to the stack space.
+ * @param stack_size Stack size in bytes.
+ * @param entry Thread entry function.
+ * @param p1 1st entry point parameter.
+ * @param p2 2nd entry point parameter.
+ * @param p3 3rd entry point parameter.
+ * @param prio Thread priority.
+ * @param options Thread options.
+ * @param delay Scheduling delay (in milliseconds), or K_NO_WAIT (for no delay).
+ *
+ * @return ID of new thread.
+ */
+extern k_tid_t k_thread_create(struct k_thread *new_thread, char *stack,
+			       size_t stack_size,
+			       void (*entry)(void *, void *, void*),
+			       void *p1, void *p2, void *p3,
+			       int prio, u32_t options, s32_t delay);
+
```

Comment:
I think it's premature to try to anticipate the SMP APIs at this time. There is too much unknown on what we need to do, the effort has not been scoped. You could probably stuff it as bits in the 'options' field FWIW.